CMakeLists.txt

```diff
@@ -0,0 +1,103 @@
+cmake_minimum_required(VERSION 3.22)
+
+include(FetchContent)
+
+project(ClpIrV1Decoder
+    LANGUAGES CXX C
+)
+
+# Enforce "Unix Makefiles" because "Ninja" does not work with the Emscripten toolchain on CXX_20 sources.
+if(NOT CMAKE_GENERATOR MATCHES "Unix Makefiles")
+    message(FATAL_ERROR "This project is intended to be built with Unix Makefiles on a Unix system.")
+endif()
+
+# Enable compile commands by default if the generator supports it.
+if (NOT CMAKE_EXPORT_COMPILE_COMMANDS)
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE BOOL
+        "Enable/Disable output of compile commands during generation." FORCE)
+endif()
+
+# Set the default build type to Release if not specified
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    message(STATUS "Setting build type to 'Release' as none was specified.")
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build (Debug, Release, RelWithDebInfo, MinSizeRel)." FORCE)
+endif()
+
+# Download and extract Boost
+set(BOOST_VERSION "1.85.0" CACHE STRING "Version of Boost library to be used when compiling CLP sources.")
+string(REPLACE "." "_" BOOST_ARCHIVE_VERSION_PART ${BOOST_VERSION})
+
+set(BOOST_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/build/wasm/boost" CACHE STRING
+    "Directory containing Boost sources, fetched from official release if non existent.")
+if (NOT EXISTS ${BOOST_SRC_DIR})
+    FetchContent_Declare(
+        Boost
+        URL "https://archives.boost.io/release/${BOOST_VERSION}/source/boost_${BOOST_ARCHIVE_VERSION_PART}.tar.gz"
+        SOURCE_DIR ${BOOST_SRC_DIR}
+    )
+    message(STATUS "Fetching Boost from official release.")
+    FetchContent_MakeAvailable(Boost)
+endif()
+
+set(BIN_NAME "ClpIrV1Decoder" CACHE STRING "Binary name for the generated .js file and .wasm file of CLP IR V1 Decoder.")
+add_executable(${BIN_NAME})
+
+target_compile_features(${BIN_NAME}
+    PRIVATE
+    cxx_std_20
+)
+
+set(CMAKE_EXECUTABLE_SUFFIX ".js")
+
+target_link_options(${BIN_NAME} PRIVATE
+    "-s EXPORTED_FUNCTIONS=['_malloc','_free']"
+    "-s MODULARIZE=1"
+    "-s EXPORT_ES6=1"
+    "-s ENVIRONMENT=worker"
+    "-lembind"
+    "-s ALLOW_MEMORY_GROWTH=1"
+    "-s WASM_BIGINT"
+    "-flto"
+    "--closure 1"
+)
```

Comment:
I believe none of the link options were set properly due to the spaces inside each option string. Note that the above syntax generates a list, and when it's being interpreted as a string, we get something like `-s EXPORTED_FUNCTIONS=['_malloc','_free'];-s MODULARIZE=1`. Note how the spaces would cause confusion to the linker (everything after the first space just appear as redundant parameters.)

Sorry for giving you a non-working suggestion in the first place.

Let's do a few improvements:
1. Remove the quotes for the string in the list as those seem unnecessary and only cause confusions.
2. Set up debug / release specific link flags to:
   1. Speed up compile time / runtime.
   2. Set the `ENVIRONMENT` to `node` rather than `worker` so that the generated binary can be run in Node.js for easier debugging.
2. For setting up libraries to be statically linked, we should use `target_link_libraries` because `-lembind` does not seem to work well with `target_link_options`.
4. For the options that take a parameter, remove the space:
   1. For the Boolean options `-s MODULARIZE=1` -> `-sMODULARIZE` or the `"1"` would be intepreted as a string if we do `-sMODULARIZE=1`.
   2. For the string options we simply replace the space with `=`.

```suggestion
if (CMAKE_BUILD_TYPE MATCHES "Release")
    set(EXTRA_LINKER_FLAGS
        -flto
        --closure=1
        -sENVIRONMENT=worker
        CACHE STRING "Extra liner flags for CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}" FORCE
    )
    message("CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}: Enabling optimization linker flags ${EXTRA_LINKER_FLAGS}")
else ()
    set(EXTRA_LINKER_FLAGS
        -sENVIRONMENT=node
        CACHE STRING "Extra liner flags for CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}" FORCE
    )
    message("CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}: Enabling debug flags linker for ${EXTRA_LINKER_FLAGS}")
endif ()
target_link_options(${BIN_NAME} PRIVATE
    ${EXTRA_LINKER_FLAGS}
    -sALLOW_MEMORY_GROWTH
    -sEXPORT_ES6
    -sMODULARIZE
    -sWASM_BIGINT
)
target_link_libraries(${BIN_NAME}
    PRIVATE
    embind
)
```